os/base.h

```diff
@@ -0,0 +1,10 @@
+// Stub out Apple internal header file to redirect to linux_base.h.
```

Comment:
this is problematic, OSX and iOS already have a public header with this name from a different project and this will conflict 

Either we have to conditionalize the install of this header based on platform (and fully replicate its contents from OSX), or we have to conditionalize its inclusion.

an `AC_CHECK_HEADERS([os/base.h])` test in configure.ac would be the standard way to do the latter, but since most of the `#include <os/base.h>` are in API headers, that is not really an option either (since we don't want to install config.h on OSX)

the best option for now may be to move the various `#include <os/base.h>` into `#ifdef __APPLE__` conditionals and in the `#else` case do the `#include <os/linux_base.h>` (or possibly include an intermediate header that does non-APPLE platform disambiguation, though based on what ?)